### PR TITLE
#437 Enhancement : Cat vs Dog Classification

### PR DESCRIPTION
**Pull request by PrashantVIT1**
## Related Issue 

- In the cat vs dog classification the previous working model was only able to predict with 89% accuracy but the enhanced model is able to predict above 92% using 4 different independent models.

- [x] DevIncept Participant
- [x] Contributor

Closes: #437

### In Cat vs Dog classification I have uploaded the enhancement folder and all changes in it.

Give a clear description of what have you added or modifications made

## Type of change

What sort of change have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

It has been tested by using the test dataset and while test it was found that the model was more capable of predicting cats than dogs. To resolve the issue here 4 independent models have been used.
Also have been tested on some random images from internet
dog 219

By creating confusion matrics

## Checklist:

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added things that prove my fix is effective or that my feature works.
- [ ] Any dependent changes have been merged and published in downstream modules.

## Screenshots (Only for Front End and UI/UX Designers)

 Original           | Updated
 :--------------------: |:--------------------:
 Original Screenshot | Updated Screenshot |
 
cat 4003
cat
dog 000

### CODE REVIEW

**Comment by PrashantVIT1** [comment]
Please comment if any chances are required

**Comment by PrashantVIT1** [comment]
please comment if any changes are required

**Review by prathimacode-hub — CHANGES_REQUESTED** [review]
Changes to be done:
Dataset folder should be as "Dataset"
Model folder should be as "Model"
Elaborate your README.md file and it's better if you do this classification using multiple algorithms. It needs more details.

Kindly do these changes.
@PrashantVIT1

**Comment by PrashantVIT1** [comment]
# Thank You!!
Maam will it be ok if I don't rename them because it takes a lot of time to upload and many times it fails in between the upload process and also in the previous work they have used the same nomanclature. So I thought it's part of the format.
Yes, I will definitely elaborate on my README.md file but using a different algorithm will be like starting from the beginning and all the effort that I have taken for testing this will get wasted. Also, I can add a confusion matrix to make the test result more clear.
And image related to neural network architecture for more clarity.
 **_(For the testing of the results I had also used some random images that are out of the labeled dataset and It was producing correct results)_**
##### **___Will it be Ok ma'am ___**

**Comment by prathimacode-hub** [comment]
> # Thank You!!
> Maam will it be ok if I don't rename them because it takes a lot of time to upload and many times it fails in between the upload process and also in the previous work they have used the same nomanclature. So I thought it's part of the format.
> Yes, I will definitely elaborate on my README.md file but using a different algorithm will be like starting from the beginning and all the effort that I have taken for testing this will get wasted. Also, I can add a confusion matrix to make the test result more clear.
> And image related to neural network architecture for more clarity.
> **_(For the testing of the results I had also used some random images that are out of the labeled dataset and It was producing correct results)_**
> 
> ##### **___Will it be Ok ma'am ___**

Ok, that's upto you. But all followed the same project structure. You will be the one odd out. Fine. @PrashantVIT1

**Comment by prathimacode-hub** [comment]
> # Thank You!!
> Maam will it be ok if I don't rename them because it takes a lot of time to upload and many times it fails in between the upload process and also in the previous work they have used the same nomanclature. So I thought it's part of the format.
> Yes, I will definitely elaborate on my README.md file but using a different algorithm will be like starting from the beginning and all the effort that I have taken for testing this will get wasted. Also, I can add a confusion matrix to make the test result more clear.
> And image related to neural network architecture for more clarity.
> **_(For the testing of the results I had also used some random images that are out of the labeled dataset and It was producing correct results)_**
> 
> ##### **___Will it be Ok ma'am ___**

Kindly do the changes. @PrashantVIT1

**Comment by PrashantVIT1** [comment]
ok I will make changes as soon as possible I had some college work
so I was busy.

**Comment by PrashantVIT1** [comment]
Maam it was showing accuracy above 96% when I was training it but when I checked its performance using actual images it was predicting correctly only on 67% of test cases. So I think may it's not working.

**Comment by prathimacode-hub** [comment]
> Maam it was showing accuracy above 96% when I was training it but when I checked its performance using actual images it was predicting correctly only on 67% of test cases. So I think may it's not working.

Oh, so what's the next step you want to take? @PrashantVIT1

**Comment by PrashantVIT1** [comment]
I have done a project in python its name is "Standardized number recognition without using ML". It works on feature detection and also has a visualization of steps that the algorithm is taking to complete the task and I have used NumPy and OpenCV. So can you suggest where I can contribute?

**Comment by PrashantVIT1** [comment]
Also the while it was building it was showing me an accuracy of about 96% but when I checked it over my test dataset it only works well for 67.5% of the time. So it is showing is good accuracy but not performing well. So can you suggest something to me, mam?
I don't have any other idea to do in this project.
but if u like I can add the code for performance check.

**Comment by prathimacode-hub** [comment]
I don't get you, what you want to say. Does it mean that you don't want to do any changes over the PR? 

And about next PR you said, if you had done, share the link to the project. So that I can see if that can fit here. @PrashantVIT1

**Comment by PrashantVIT1** [comment]
Yes ma'am I don't know why it's not performing well even it is showing it's 96% accurate.
so here is the link to my python project https://github.com/PrashantVIT1/Standardized-Number-Recognition/tree/main
Please check this project ma'am.

**Comment by prathimacode-hub** [comment]
Ok, have you tried to optimize it or hyperparamater tuning? @PrashantVIT1

**Comment by PrashantVIT1** [comment]
yes, ma'am, I tried hyperparameter tuning. 
Can I put performance testing code, ma'am, because it's not there in the project?

**Comment by PrashantVIT1** [comment]
have you seen this project it is related to optical character recognition 
https://github.com/PrashantVIT1/Standardized-Number-Recognition/tree/main

**Comment by prathimacode-hub** [comment]
> yes, ma'am, I tried hyperparameter tuning.
> Can I put performance testing code, ma'am, because it's not there in the project?

Yes, you can put it off. That's Ok. Whatever you tried is fine even though you couldn't achieve the accuracy expect. Next time you can find more ways to make it better. @PrashantVIT1

**Comment by prathimacode-hub** [comment]
> have you seen this project it is related to optical character recognition
> https://github.com/PrashantVIT1/Standardized-Number-Recognition/tree/main

Yes, I did check it. You can do the same but with ML. Without ML, this repo won't be able to justify. @PrashantVIT1

**Comment by prathimacode-hub** [comment]
Have you done the changes for this PR? @PrashantVIT1

**Comment by PrashantVIT1** [comment]
No, I think in cat vs dog classifier the part I was working on was not working as it should I don't know the exact reason even tried hyper parameter tuning but result the same. So can you suggest me something.

**Review by prathimacode-hub — CHANGES_REQUESTED** [review]
The thing is the libraries are not imported properly. That's the main reason for it to not function properly. Remove all the libraries and try reinstalling again with right ones. This would work. And also change the PR files according to project structure. @PrashantVIT1

**Comment by prathimacode-hub** [comment]
Also 9th is the last date to accept PR's. If you wouldn't do it by then, obviously you can submit later but points won't be counted towards DCP program. So kindly do it by then. @PrashantVIT1

**Comment by prathimacode-hub** [comment]
Do it by today. @PrashantVIT1

**Comment by prathimacode-hub** [comment]
Sorry time is up and unfortunately i have to take off the label. If you want  to continue further and address the changes, make sure you complete it in 2days. It's not right to keep PR open for so long. Its already been 23 days. @PrashantVIT1
